programs/lib/include/flexpret_types.h

```diff
@@ -18,4 +18,31 @@ typedef enum {
     FP_OUT_OF_MEMORY = 3
 } fp_ret_t;
 
+/**
+ * Use this for initializing arrays with varying size depending on the NUM_THREADS
+ * macro. Example use case:
+ * 
+ * static bool need_init[NUM_THREADS] = ARRAY_INITIALIZER(true);
+ * 
+ * static uint8_t nbytes_left[NUM_THREADS] = ARRAY_INITIALIZER(5);
+ * 
+ */
+#if NUM_THREADS == 1
+    #define ARRAY_INITIALIZER(val) { val }
```

Comment:
I would rename it to: `THREAD_ARRAY_INITIALIZER` or something to communicate that we are initializing an array of length equal to number of threads